### ISSUE TITLE
Add an option to print full failure messages

### ISSUE THREAD

**Issue by wallrj** [jonathanj/eliottree#12]
In this example, the interesting error is

```
exceptions.IOError: [Errno 2] No such file or directory: '/etc/flocker/agent.yml'\n"
```

But Eliot tree doesn't show it.

```
(1988)(flocker-ca-output-FLOC-1988 ? =)[~/.../HybridLogic/flocker]$ flocker-dataset-agent  > log
(1988)(flocker-ca-output-FLOC-1988 ? =)[~/.../HybridLogic/flocker]$ cat log
{"task_uuid": "cdeb220d-7605-4d5f-8341-1a170222e308", "error": false, "timestamp": 1432731519.112539, "message": "Log opened.", "message_type": "twisted:log", "task_level": [1]}
{"status": 1, "zfs_command": "zfs set readonly=on flocker", "output": "[Errno 2] No such file or directory", "timestamp": 1432731519.117719, "message_type": "filesystem:zfs:error", "task_uuid": "cdeb220d-7605-4d5f-8341-1a170222e308", "task_level": [2]}
{"status": 1, "zfs_command": "zfs set canmount=off flocker", "output": "[Errno 2] No such file or directory", "timestamp": 1432731519.122177, "message_type": "filesystem:zfs:error", "task_uuid": "cdeb220d-7605-4d5f-8341-1a170222e308", "task_level": [3]}
{"task_uuid": "cdeb220d-7605-4d5f-8341-1a170222e308", "error": true, "timestamp": 1432731519.123826, "message": "Unhandled Error\nTraceback (most recent call last):\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/node/script.py\", line 404, in flocker_dataset_agent_main\n    options=options,\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/common/script.py\", line 187, in main\n    self._react(run_and_log, [], _reactor=self._reactor)\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/internet/task.py\", line 875, in react\n    finished = main(_reactor, *argv)\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/common/script.py\", line 178, in run_and_log\n    d = maybeDeferred(self.script.main, reactor, options)\n--- <exception caught here> ---\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/internet/defer.py\", line 140, in maybeDeferred\n    result = f(*args, **kw)\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/volume/service.py\", line 452, in main\n    return self._volume_script.main(reactor, options, service)\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/node/script.py\", line 366, in main\n    configuration = yaml.safe_load(agent_config.getContent())\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/python/filepath.py\", line 290, in getContent\n    fp = self.open()\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/python/filepath.py\", line 845, in open\n    return open(self.path, mode + 'b')\nexceptions.IOError: [Errno 2] No such file or directory: '/etc/flocker/agent.yml'\n", "message_type": "twisted:log", "task_level": [4]}
{"task_uuid": "cdeb220d-7605-4d5f-8341-1a170222e308", "error": true, "timestamp": 1432731519.124293, "message": "main function encountered error\nTraceback (most recent call last):\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/node/script.py\", line 404, in flocker_dataset_agent_main\n    options=options,\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/common/script.py\", line 187, in main\n    self._react(run_and_log, [], _reactor=self._reactor)\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/internet/task.py\", line 875, in react\n    finished = main(_reactor, *argv)\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/common/script.py\", line 178, in run_and_log\n    d = maybeDeferred(self.script.main, reactor, options)\n--- <exception caught here> ---\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/internet/defer.py\", line 140, in maybeDeferred\n    result = f(*args, **kw)\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/volume/service.py\", line 452, in main\n    return self._volume_script.main(reactor, options, service)\n  File \"/home/richard/projects/HybridLogic/flocker/flocker/node/script.py\", line 366, in main\n    configuration = yaml.safe_load(agent_config.getContent())\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/python/filepath.py\", line 290, in getContent\n    fp = self.open()\n  File \"/home/richard/.virtualenvs/1988/lib/python2.7/site-packages/twisted/python/filepath.py\", line 845, in open\n    return open(self.path, mode + 'b')\nexceptions.IOError: [Errno 2] No such file or directory: '/etc/flocker/agent.yml'\n", "message_type": "twisted:log", "task_level": [5]}
{"task_uuid": "cdeb220d-7605-4d5f-8341-1a170222e308", "error": false, "timestamp": 1432731519.12443, "message": "Main loop terminated.", "message_type": "twisted:log", "task_level": [6]}
(1988)(flocker-ca-output-FLOC-1988 ? =)[~/.../HybridLogic/flocker]$ eliot-tree log 
cdeb220d-7605-4d5f-8341-1a170222e308
    +-- twisted:log@1/None
        |-- error: False
        |-- message: Log opened.
        |-- message_type: twisted:log
        `-- timestamp: 2015-05-27 13:58:39.112539
    +-- filesystem:zfs:error@2/None
        |-- message_type: filesystem:zfs:error
        |-- output: [Errno 2] No such file or directory
        |-- status: 1
        |-- timestamp: 2015-05-27 13:58:39.117719
        `-- zfs_command: zfs set readonly=on flocker
    +-- filesystem:zfs:error@3/None
        |-- message_type: filesystem:zfs:error
        |-- output: [Errno 2] No such file or directory
        |-- status: 1
        |-- timestamp: 2015-05-27 13:58:39.122177
        `-- zfs_command: zfs set canmount=off flocker
    +-- twisted:log@4/None
        |-- error: True
        |-- message: Unhandled Error...
        |-- message_type: twisted:log
        `-- timestamp: 2015-05-27 13:58:39.123826
    +-- twisted:log@5/None
        |-- error: True
        |-- message: main function encountered error...
        |-- message_type: twisted:log
        `-- timestamp: 2015-05-27 13:58:39.124293
    +-- twisted:log@6/None
        |-- error: False
        |-- message: Main loop terminated.
        |-- message_type: twisted:log
        `-- timestamp: 2015-05-27 13:58:39.124430

```

It'd be nice if it had an option to show full un-elipsized (is that the term) message field contents.
Something like `journalctl` s `--full` option, maybe?

Or have a way to show `twisted:log@5/None` in full?

**Comment by jonathanj** [comment]
@wallrj Can you give the code referenced by PR a go with the `--no-truncate` option and perhaps comment on it with your findings?

**Comment by jonathanj** [comment]
Closed by c75c1c293b9556c50b7a727aab15c0f386c0e30d.
